### PR TITLE
Update UKXattrMetadataStore.[mh]

### PR DESCRIPTION
**Pull request by MaddTheSane**
Add more Swift-friendly methods.

### CODE REVIEW

**Comment by uliwitness** [comment]
Hi, thanks for the merge request, I'd love to merge those changes, but I'd need a few changes.

File: UKXattrMetadataStore.h [hunk 1]
```diff
@@ -1,37 +1,22 @@
 //
 //  UKXattrMetadataStore.h
 //  BubbleBrowser
+//	LICENSE: MIT License
```
Comment (uliwitness) [inline UKXattrMetadataStore.h]: Nope. This license is not the MIT license. MIT requires credit, zlib doesn't. Can you please undo this change?

Comment (MaddTheSane) [reply UKXattrMetadataStore.h]: Yeah. Sorry about that.

File: UKXattrMetadataStore.m [hunk 2]
```diff
@@ -43,31 +29,25 @@ @implementation UKXattrMetadataStore
...
++(NSArray*) allKeysAtPath: (NSString*)path traverseLink:(BOOL)travLnk
 {
-	NSMutableArray*	allKeys = [NSMutableArray array];
+	NSMutableArray<NSString*>*	allKeys = [NSMutableArray array];
 	size_t dataSize = listxattr( [path fileSystemRepresentation],
 								NULL, ULONG_MAX,
 								(travLnk ? 0 : XATTR_NOFOLLOW) );
 	if( dataSize == ULONG_MAX )
-		return allKeys;	// Empty list.
+		return nil;	// Empty list.
```
Comment (uliwitness) [inline UKXattrMetadataStore.m]: Can't return NIL here, your header has ASSUME_NONNULL on. Also, why wouldn't you return an empty array here if it has no xattrs?

File: UKXattrMetadataStore.m [hunk 3]
```diff
@@ -103,11 +93,24 @@ +(void)	setObject: (id)obj forKey: (NSString*)key atPath: (NSString*)path traver
...
 	{
-		[errMsg autorelease];
 		[NSException raise: @"UKXattrMetastoreCantSerialize" format: @"%@", errMsg];
 	}
 	else
-		[[self class] setData: plistData forKey: key atPath: path traverseLink: travLnk];
+		[self setData: plistData forKey: key atPath: path traverseLink: travLnk error: NULL];
+}
+
++(BOOL)	setObject: (id)obj forKey: (NSString*)key atPath: (NSString*)path traverseLink:(BOOL)travLnk error:(NSError**)error
```
Comment (uliwitness) [inline UKXattrMetadataStore.m]: I'd rename this to `setPlistXML:`. `setObject:` implies it might be the right thing to use for an `NSDate` or `NSString`.

File: UKXattrMetadataStore.m [hunk 4]
```diff
@@ -159,21 +194,31 @@ +(NSMutableData*)	dataForKey: (NSString*)key atPath: (NSString*)path traverseLin
...
+	NSError		*err = nil;
+	id obj = [self objectForKey:key atPath:path traverseLink:travLnk error:&err];
+	if (!obj) {
+		[NSException raise:@"UKXattrMetastoreCantUnserialize" format: @"%@", err];
+	}
+	
+	return obj;
+}
+
++(nullable id) objectForKey: (NSString*)key atPath: (NSString*)path
```
Comment (uliwitness) [inline UKXattrMetadataStore.m]: `objectForPlistXMLInKey:` ?

**Comment by uliwitness** [comment]
So, I can make all the changes I mention when I merge it, but I'd need you to confirm that you're OK with your changes being licensed under zlib, not MIT, before I merge them, and I'd be curious why you think this should return NIL instead of an empty array before I change it back and functionality is lost.

File: UKXattrMetadataStore.m (hunk 3 above)
Comment (uliwitness) [reply UKXattrMetadataStore.m]: (Nevermind, this was my naming mistake from earlier on -- will fix this and leave a compatibility method for ObjC)

**Comment by uliwitness** [comment]
Thanks for the update!
